index.html

```diff
@@ -2730,35 +2758,55 @@ <h3><dfn>Writing or pushing content</dfn></h3>
                       <a>map binary data to NDEF</a>.
                     </li>
                   </ul>
-                  <dt><a>external type name</a></dt>
-                  <ul>
-                    If |record|'s <a>recordType</a> is an <a>external type</a>,
-                    <li>
-                      If |record|'s <a>data</a> is of type {{NDEFMessageInit}},
-                      then return the result of running the
-                      <a>create NDEF message</a> given |record|'s <a>data</a>.
-                    </li>
-                    <li>
-                      Otherwise, <a>map external data to NDEF</a> given |record|.
-                    </li>
-                  </ul>
-                  <dt><a>local type name</a></dt>
+                  <dt>unmatched type</dt>
                   <ul>
-                    If |record|'s <a>recordType</a> is a <a>local type</a> and if
-                    |record| is a payload to another <a>NDEF</a> record,
                     <li>
-                      If |record|'s <a>data</a> is of type {{NDEFMessageInit}},
-                      then return the result of running the
-                      <a>create NDEF message</a> given |record|'s <a>data</a>.
+                    <dl>
+                      <dt>Check for <a>external type</a>.</dt>
+                      Let |name:string| be the result of running the
+                      <a>validate external type</a> steps on |record|'s
+                      <a>recordType</a>. If |name| is `false`, continue with
+                      the next step. Otherwise,
+                      <ul>
+                        <li>
+                          Let |record|'s <a>recordType</a> be |name:string|.
+                        </li>
+                        <li>
+                          If |record|'s <a>data</a> is of type
+                          {{NDEFMessageInit}}, then return the result of running
+                          the <a>create NDEF message</a> given |record|'s
+                          <a>data</a>.
+                        </li>
+                        <li>
+                          Otherwise, <a>map external data to NDEF</a> given |record|.
+                        </li>
+                      </ul>
+                    </dl>
                     </li>
                     <li>
-                      Otherwise, <a>map local type to NDEF</a> given |record|.
-                    </li>
-                  </ul>
-                  <dt>unmatched type</dt>
-                  <ul>
+                    <dl>
+                      <dt>Check for <a>local type</a>.</dt>
```

Comment:
Is `Check for` enough to know whether or not a recordType is "local type" or "external type"?

Reply:
Shall we link to "Validating" sections?

Reply:
This was just the title, the whole paragraph was about that check, but now I removed the title.